controller/aks-cluster-config-handler.go

```diff
@@ -184,7 +185,25 @@ func (h *Handler) recordError(onChange func(key string, config *aksv1.AKSCluster
 	return func(key string, config *aksv1.AKSClusterConfig) (*aksv1.AKSClusterConfig, error) {
 		var err error
 		var message string
-		config, err = onChange(key, config)
+
+		err = retry.RetryOnConflict(retry.DefaultRetry, func() error {
+			// AKS config is being deleted
+			if config == nil {
+				return nil
+			}
+
+			conf, err := h.aksCC.Get(config.Namespace, config.Name, metav1.GetOptions{})
+			if err != nil {
+				return err
+			}
+
+			conf, err = onChange(key, conf)
+			if err == nil {
```

Comment:
I think we can update `config` regardless of whether `err `is nil to reflect any changes made by `onChange`, even in error scenarios.

```suggestion
			config = conf
```

Reply:
Unfortunately when a conflict occurs, the config object returned from `onChange` is empty/non-nil which causes the original bug. That is the sole reason for checking if `err` is nil.